File: 1013-fibonacci-number/solution.c


// int fib(int n){
//     int a=0;
//     int b=1;
//     int result;
//     int i=2;
//     if(n==0){
//         return 0;
//     }
//     if(n==1){
//         return 1;
//     }
//     else{
//         return fib(n - 1) + fib(n - 2);
//     }

// }

int fib(int n){
    int a=0;
    int b=1;
    int result;
    int i=2;
    if(n==0){
        return 0;
    }
    if(n==1){
        return 1;
    }
    while(i<=n){
        result=a+b;
        a=b;
        b=result;
        i++;
    }
    return result;

}

File: 1013-fibonacci-number/solution.java
class Solution {
    public int sol(int n, int dp[]){
        if(n==0||n==1){
            return n;
        }
        return sol(n-1,dp)+sol(n-2,dp);
    }

    public int fib(int n) {
        int dp[]= new int[n+1];
        Arrays.fill(dp,-1);
        return sol(n,dp);
    }
}
